### PR TITLE
feat: changes to support data, timestamp and arrays in IT tests

### PR DESCRIPTION
**Pull request by asthamohta**

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java [hunk 1]
```diff
@@ -771,7 +767,7 @@ public void bindNumericArray_doesNotPreservePrecision() {
 
     Struct row =
         execute(
-            Statement.newBuilder("SELECT @v").bind("v").toNumericArray(asList(b1, b2, null)),
+            Statement.newBuilder(selectValueQuery).bind("p1").toNumericArray(asList(b1, b2, null)),
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java:770]: This test case seem to be disabled for PostgreSQL still. Is that intentional?
The same also seems to be the case for other numeric array tests.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java [hunk 2]
```diff
@@ -429,6 +522,59 @@ public void testReadNullValuesInArrays() {
     }
   }
 
+  @Test
+  public void testReadNullValuesInArraysPostgreSQL() {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java:526]: Why do we need separate tests for GoogleSQL and PostgreSQL in this case? They seem to be equal. Or am I missing something?

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java [hunk 3]
```diff
@@ -615,9 +793,16 @@ public void testReadFloat64LiteralsGoogleStandardSQL() {
   public void testReadFloat64LiteralsPostgreSQL() {
     Assume.assumeTrue(dialect.dialect == Dialect.POSTGRESQL);
     try (ResultSet resultSet =
-        databaseClient.singleUse().executeQuery(Statement.of("SELECT 10.0 AS float64"))) {
+        databaseClient
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java:796]: Why does this need a separate test case?

**Review by olavloite — APPROVED** [review]
Looks good to me, with a couple of minor questions.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java (hunk 2 above)
Comment (asthamohta) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java:526]: structs and json are not supported and column names are different as PG requires explicit quotes for case

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java (hunk 3 above)
Comment (asthamohta) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITResultSetGetValue.java:796]: same as above

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java (hunk 1 above)
Comment (asthamohta) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITQueryTest.java:770]: missed those, adding it

**Review by ansh0l — APPROVED** [review]
lgtm

**Pull request merged**